test/e2e/operator_test.go

```diff
@@ -1076,6 +1077,162 @@ func TestContainerLogging(t *testing.T) {
 	}
 }
 
+// NOTE: This test will mutate the default ingresscontroller and the
+// "cluster" ingress configuration.
+func TestRouteHTTP2EnableAndDisable(t *testing.T) {
+	routerDeployTimeout := 1 * time.Minute
+
+	assertPreConditions := func(t *testing.T, ic *operatorv1.IngressController, cfg *configv1.Ingress, d *appsv1.Deployment) {
+		t.Helper()
+		if ingress.HTTP2IsDisabledByAnnotation(ic.Annotations) {
+			t.Fatal("expected ingress controller to have HTTP/2 enabled by default")
+		}
+		if ingress.HTTP2IsDisabledByAnnotation(cfg.Annotations) {
+			t.Fatal("expected infrastructure config to have HTTP/2 enabled by default")
+		}
+		if len(d.Spec.Template.Spec.Containers) < 1 {
+			t.Fatal("expected routerDeployment to have at least one container")
+		}
+		if http2IsDisabledInEnv(d.Spec.Template.Spec.Containers[0].Env) {
+			t.Fatal("expected routerDeployment to have http/2 enabled by default")
+		}
+	}
+
+	refreshObjects := func() (*operatorv1.IngressController, *configv1.Ingress, *appsv1.Deployment) {
```

Comment:
Maybe `t.Helper()` here too?

Also, are you defining these functions inside `TestRouteHTTP2EnableAndDisable` in order to group them, or is there some other reason?  You could pass in `t` (as you already do with `assertPreConditions`), and I don't see any other free variables.  That said, I'm not opposed to defining the functions this way for grouping if that is the style you prefer.

All that said, it might be worth moving the test and its helpers to a separate file as `operator_test.go` has grown rather large.

Reply:
The purpose wasn't necessarily to group them but to contain their scope and their semantics to just the function under test. They call fatal (now with the helper) which may be undesriable as free standing functions. I can move refreshObjects() out but assertPreConditions() really is for that test function alone.

t.Helper - done.